### PR TITLE
[Core] OctreeBinary cleanup

### PR DESCRIPTION
**Pull request by miguelmaso**
**📝 Description**
Related to #9217 . I wanted to extend FindIntersectedGeometricalObjectsProcess, which uses the OctreeBinary. Then I realized that there is some duplication of code.
Maybe there are more unused methods, but I don't know if can be safely removed.

To merge after #9217

**🆕 Changelog**
- Remove commented code in `octree_binary`
- Remove unused `ifdef KRATOS_DEPENDENT` from `octree_binary`
- Remove the implementation of `OctreeBinary.HasIntersection` by calling `Geometry.HasIntersection`
- Override `Line3D3.HasIntersection`

### CODE REVIEW

**Review by rubenzorrilla — COMMENTED** [review]
All the sensitive changes in here are to be discussed in #9217. Aside of these, the rest of modifications in this PR are clean up with no behavior change. Thanks!

File: kratos/geometries/line_3d_2.h [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include "geometries/geometry.h"
 #include "integration/line_gauss_legendre_integration_points.h"
 #include "integration/line_collocation_integration_points.h"
+#include "utilities/intersection_utilities.h"
```
Comment (philbucher) [inline kratos/geometries/line_3d_2.h:28]: this includes `node.h` (which it does not use)
@miguelmaso could you please clean the includes to keep the compile times and the dependencies low?
Thx!

**Comment by miguelmaso** [comment]
now I think its ready to merge

File: kratos/geometries/line_3d_2.h (hunk 1 above)
Comment (miguelmaso) [reply kratos/geometries/line_3d_2.h:28]: done

**Comment by philbucher** [comment]
ok for me

as it is Core I leave approval to @rubenzorrilla

**Comment by miguelmaso** [comment]
ping @rubenzorrilla

**Review by rubenzorrilla — APPROVED** [review]
Nice clean up. Thanks @miguelmaso !!!

**Pull request merged**